### ISSUE TITLE
Make FMS Tests accessible via CMake

### ISSUE THREAD

**Issue by mathomp4** [NOAA-GFDL/FMS#827]
**Is your feature request related to a problem? Please describe.**

As you might know, I've been looking at building GEOS with recent FMS (see #796 and #806) and I've had some success with the CMake build. Huzzah! But recently, we remembered #54 and FMS' use of Cray Pointers which is one of our stumbling blocks in using NAG with GEOS (one of many, but one). 

So, I cloned up FMS, added an option in CMake to add `-DPORTABLE_KINDS`, pointed to a NAG netCDF, and eventually got to:
```
Error: /discover/nobackup/mathomp4/BuildFMS/FMS/mpp/include/mpp_comm_mpi.inc, line 364: Syntax error
       detected at POINTER@(
```
as expected. 

Thus, we could have something to work on fixing up the Cray pointers, but then I realized, we might get it to build at some point, but at the moment our only real FMS tester is GEOS. And since it has many, many places needing fixing for NAG as well (hello `real*8`), it's not ideal.

I know that FMS does have the `test_fms/` suite but my reading of it and `CMakeLists.txt` looks like it is Autotools-only at the moment, right?

**Describe the solution you'd like**

The ability to run the `test_fms/` suite when building with CMake.

**Describe alternatives you've considered**

Well, we could do our work with Autotools. My only concern is might need to depend on the knowledge and good graces of @rem1776. I'm fair at CMake, but my Autotools hacking is lacking for sure. I always shiver a bit when I realize I have to edit a `configure.ac` or `Makefile.in`.

**Comment by rem1776** [comment]
@mathomp4 You are correct, the test suite is only through autotools currently. I believe Cmake should be able to run the test suite since it's essentially just shell scripts, but there may be some difficulties from the differences in how they test/directory structure. I'll bring it up with our team and see if someone may have a bit more experience with cmake testing.

**Comment by mathomp4** [comment]
@rem1776 Note that it's possible if this effort moves up in our priorities, we might take it on. We do have some experience with CMake and CTest. As you say, it is just shell scripts and making the test programs! 

But thanks for checking with your team.

**Comment by rem1776** [comment]
@mathomp4 (or any other motivated parties) Feel free to take this on if you feel this would be beneficial. Since we're building with CMake in spack currently it would be great to get some actual run checks with the resulting library from the cmake build.
